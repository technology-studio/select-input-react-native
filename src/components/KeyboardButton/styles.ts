import {
  StyleSheet,
} from 'react-native'

export const styles = StyleSheet.create({
  buttontext: {
    color: '#006BFF',
    padding: 8,
  },
})
