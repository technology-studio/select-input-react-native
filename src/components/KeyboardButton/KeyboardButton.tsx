import React, { Component } from 'react'
import {
  type StyleProp,
  Text, TouchableOpacity,
} from 'react-native'

import { styles } from './styles'

type Props = {
  onPress: () => void,
  text: string,
  textAlign: 'center' | 'left' | 'right',
  textStyle?: StyleProp<Record<string, unknown>>,
}

export class KeyboardButton extends Component<Props> {
  render (): JSX.Element {
    const { onPress, text, textAlign, textStyle } = this.props

    return (
      <TouchableOpacity onPress={onPress}>
        <Text
          allowFontScaling={false}
          style={[styles.buttontext, textStyle, { textAlign }]}
        >
          {text}
        </Text>
      </TouchableOpacity>
    )
  }
}
