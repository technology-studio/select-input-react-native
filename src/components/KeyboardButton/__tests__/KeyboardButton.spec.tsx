import React from 'react'
import renderer from 'react-test-renderer'

import { KeyboardButton } from '../KeyboardButton'

const props = {
  onPress: jest.fn(),
  text: 'buttonText',
  textAlign: 'left' as const,
  textStyle: {
    color: '#ABCDEF',
    fontSize: 13,
  },
}

describe('KeyboardButton', () => {
  test('renders correctly with given props', () => {
    const keyboardButton = renderer.create(<KeyboardButton {...props} />)
    expect(keyboardButton).toMatchSnapshot()
  })
})
