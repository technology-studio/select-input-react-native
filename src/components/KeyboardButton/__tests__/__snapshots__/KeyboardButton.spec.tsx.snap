// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardButton renders correctly with given props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#006BFF",
          "padding": 8,
        },
        Object {
          "color": "#ABCDEF",
          "fontSize": 13,
        },
        Object {
          "textAlign": "left",
        },
      ]
    }
  >
    buttonText
  </Text>
</View>
`;
